projects/gateway2/translator/listener/gateway_listener_translator.go

```diff
@@ -786,3 +788,46 @@ func makeVhostName(
 ) string {
 	return utils.SanitizeForEnvoy(ctx, parentName+"~"+domain, "vHost")
 }
+
+// AppendHealthCheckListener appends a readiness listener with the health check plugin defined on it
+// and returns the updated listener list
+func AppendHealthCheckListener(listeners []*v1.Listener) []*v1.Listener {
+	return append(listeners, GenerateHealthCheckListener())
+}
+
+// GenerateHealthCheckListener returns a readiness listener with the health check plugin defined on it
+func GenerateHealthCheckListener() *v1.Listener {
```

Comment:
Do we have a way of validating that this generates the same config as the removed Helm? Or are the passing tests enough validation?

Reply:
The passing tests should do. Additionally the changes to the test files (https://github.com/solo-io/gloo/pull/10542/files#diff-1b1edb892a960bc5d498be228066a8c5eaa997b92dfb62a9c297e03c6b0fe07e) show the envoy listener generated